custom/enikshay/tasks.py

```diff
@@ -624,6 +646,16 @@ def check_and_return(self, update_dict):
         else:
             return None
 
+    def ledger_updates(self):
+        _updates = []
+        for dose_date, dose_status in self.dose_status_by_date.items():
+            if dose_status.source and dose_status.value:
+                expected_value = get_expected_fixture_value(self.domain, dose_status.source, dose_status.value)
+                entry_id = ledger_entry_id_for_adherence(dose_date)
+                if ledger_needs_update(self.domain, self.episode.case_id, entry_id, expected_value):
+                    _updates.append((self.episode.case_id, entry_id, expected_value))
```

Comment:
I'm curious why you chose to pass this through as a tuple instead of just creating the `Balance` xml directly here?

Reply:
just following the same process that is being used for case updates.